website/components/MiddleTruncatedText.tsx

```diff
@@ -0,0 +1,107 @@
+import styled, { css } from "styled-components";
+
+// Adapted from "CSS-only middle truncation with ellipsis"
+// by Mark Chitty https://codepen.io/markchitty/pen/RNZbRE
+
+type MiddleTruncatedTextProps = {
+  text: string;
+  /** Minimum characters to show at the start of the text. 3 by default. */
+  startCharacters?: number;
+  /**
+   * Minimum characters to show at the end of the text. Hidden by start characters if the container
+   * does not have enough space. 35 by default.
+   */
+  endCharacters?: number;
+};
+
+const defaultProps: Partial<MiddleTruncatedTextProps> = {
+  startCharacters: 3,
+  endCharacters: 35,
+};
+
+const MiddleTruncatedTextContainer = styled.div<{ $startChars: number; $endChars: number }>`
+  display: flex;
+  flex-direction: row;
+  vertical-align: bottom;
+  white-space: nowrap;
+
+  ${(props) => {
+    // Determined experimentally. If this is too high, excess padding will appear to the right of
+    // the end text. If this is too low, the end text will be clipped.
+    const characterPxWidthToFontSizeRatio = 0.5;
+    return css`
+      --startWidth: calc(1em * ${props.$startChars + 3} * ${characterPxWidthToFontSizeRatio});
+      --endWidth: calc(1em * ${props.$endChars} * ${characterPxWidthToFontSizeRatio});
+
+      & span {
+        display: inline-block;
+        vertical-align: bottom;
+        white-space: nowrap;
+        overflow: hidden;
+      }
+
+      & span:first-child {
+        max-width: calc(100% - var(--endWidth));
+        min-width: var(--startWidth);
+        text-overflow: ellipsis;
+        overflow: hidden;
+      }
+
+      & span:last-child {
+        max-width: calc(100% - var(--startWidth));
+        direction: rtl;
+        z-index: 800;
+      }
+    `;
+  }}
+`;
+
+/**
+ * Renders text that will be truncated with ellipses in the middle of the text if there isn't
+ * enough space to show the full text. The number of characters shown at the start and end of the text are configurable.
+ *
+ * @example
+ * ```
+ * <MiddleTruncatedText text="This is a long text that will be truncated in the middle" startCharacters={3} endCharacters={5} />
+ * ```
+ * can be rendered as any of the following depending on the available space:
+ * - `This is a long text that will be truncated in the middle`
+ * - `This is a long text that will ...iddle`
+ * - `This is a long text...iddle`
+ * - `Thi...iddle`
+ * - `Thi...`
+ */
+export default function MiddleTruncatedText(inputProps: MiddleTruncatedTextProps): ReactElement {
+  const props = { ...defaultProps, ...inputProps } as Required<MiddleTruncatedTextProps>;
+
+  let startChars = Math.max(0, props.startCharacters);
+  let endChars = Math.max(0, props.endCharacters);
+  if (props.text.length <= startChars + endChars) {
+    endChars = props.text.length - startChars;
+  }
+
+  const splitIndex = Math.max(0, props.text.length - endChars);
+  const startText = props.text.slice(0, splitIndex);
+  const endText = props.text.slice(splitIndex);
```

Comment:
I used this `const endText = props.text.slice(splitIndex+3);` and it seemed to prevent characters from being partially rendered in the middle. 3 is an arbitrary number but I found characters were cut off using 1 or 2, this was the lowest that seemed to work, higher values also worked!

Reply:
I think that might also cause text to be missing characters in the middle... The full startText and endText are visible if the container is wide enough, so we'd be missing three characters from `splitIndex` to `splitIndex + 2`.